src/gflownet/data/sampling_iterator.py

```diff
@@ -143,23 +156,29 @@ def __iter__(self):
             # Sample conditional info such as temperature, trade-off weights, etc.
 
             if self.sample_cond_info:
-                cond_info = self.task.sample_conditional_information(num_offline + self.online_batch_size)
+                num_online = self.online_batch_size
+                cond_info = self.task.sample_conditional_information(
+                    num_offline + self.online_batch_size, self.train_it
+                )
+
                 # Sample some dataset data
                 mols, flat_rewards = map(list, zip(*[self.data[i] for i in idcs])) if len(idcs) else ([], [])
                 flat_rewards = (
                     list(self.task.flat_reward_transform(torch.stack(flat_rewards))) if len(flat_rewards) else []
                 )
                 graphs = [self.ctx.mol_to_graph(m) for m in mols]
                 trajs = self.algo.create_training_data_from_graphs(graphs)
-                num_online = self.online_batch_size
+
             else:  # If we're not sampling the conditionals, then the idcs refer to listed preferences
                 num_online = num_offline
                 num_offline = 0
-                cond_info = self.task.encode_conditional_information(torch.stack([self.data[i] for i in idcs]))
+                cond_info = self.task.encode_conditional_information(
```

Comment:
Yeah maybe a bit opaque right now but we simply concatenate the preferences and focus-directions together in the validation dataset. This is in part because preference-conditioning and goal-conditioning are not an either-or, we support using both of them together in some cases (e.g. fixed set of goals with dirichlet-sampled preferences). I'll add some documentation in the validation dataset constructor and in encode_conditional_information() to specify that we make these positional assumptions.